misc/makefile_help.go

```diff
@@ -0,0 +1,381 @@
+type Config struct {
+	Makefile   string         // path to the main Makefile
+	RelativeTo string         // base path for printing sub‑directory commands
+	Dirs       flagList       // additional directories to scan
+	Wildcards  nestedFlagList // wildcard values to expand '%' targets
+}
+
+// parseConfig parses flags and the single required Makefile argument.
+// Returns a non‑nil *Config on success, or an error on bad input.
+func parseConfig(args []string) (*Config, *flag.FlagSet, error) {
+	cfg := &Config{}
+	fs := flag.NewFlagSet("makefile-help", flag.ContinueOnError)
+
+	fs.StringVar(&cfg.RelativeTo, "relative-to", "", "base path for sub‑directory commands")
+	fs.StringVar(&cfg.RelativeTo, "r", "", "shorthand for --relative-to")
+	fs.Var(&cfg.Dirs, "dir", "directory to scan for Makefiles (repeatable)")
+	fs.Var(&cfg.Dirs, "d", "shorthand for --dir")
+	fs.Var(&cfg.Wildcards, "wildcard", "value to substitute for '%' in targets (repeatable)")
+	fs.Var(&cfg.Wildcards, "w", "shorthand for --wildcard")
+
+	if err := fs.Parse(args); err != nil {
+		return nil, fs, err
+	}
+	rest := fs.Args()
+	if len(rest) != 1 {
+		return nil, fs, errors.New("must specify exactly one Makefile path")
+	}
+	cfg.Makefile = rest[0]
+	info, err := os.Stat(cfg.Makefile)
+	if err != nil || info.IsDir() {
+		return nil, fs, fmt.Errorf("cannot read Makefile %q: %w", cfg.Makefile, err)
+	}
+	return cfg, fs, nil
+}
+
+// extractMakefileTargets reads filePath and returns a map of target⇒description.
+// It ignores lines not starting with a letter, without ':', or marked @LEGACY.
+// Inline comments after one or more '#'s become descriptions.
+func extractMakefileTargets(filePath string) (map[string]string, error) {
+	legacyRe := regexp.MustCompile(`#.*@LEGACY\b`)
+
+	f, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	targets := make(map[string]string)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(line) == 0 || !unicode.IsLetter(rune(line[0])) {
+			continue
+		}
+		colon := strings.IndexRune(line, ':')
+		if colon < 0 || legacyRe.MatchString(line) || (colon+1 < len(line) && line[colon+1] == '=') {
+			continue
+		}
+
+		name := line[:colon]
+		desc := ""
+		for i := colon + 1; i < len(line); i++ {
+			if line[i] == '#' && (i+1 < len(line) && line[i+1] != '#') {
+				desc = strings.TrimSpace(line[i+1:])
+				break
+			}
+		}
+		targets[name] = desc
+	}
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+	return targets, nil
+}
+
+// maxStringLength returns the length of the longest string in items.
+func maxStringLength(items []string) int {
+	maxLen := 0
+	for _, s := range items {
+		if len(s) > maxLen {
+			maxLen = len(s)
+		}
+	}
+	return maxLen
+}
+
+// maxKeyLength calculates the column width for printing keys,
+// accounting for '%' expansions if wildcards are provided.
+func maxKeyLength(keys []string, wildcards [][]string) int {
+	totalWildcards := 0
```

Comment:
This can be simplified, but I think it will be anyway if we change to `-replace`